### PR TITLE
User account deletion

### PR DESCRIPTION
**Pull request by Williams94**
Let me know what you think and if you have any suggestions. 

I'm not sure about the actual settings modal design now because there isn't the space at the bottom anymore since I added the delete button there.

Also went for a very simple user confirmation when trying to delete, but it could be made slightly more serious.

### CODE REVIEW

File: app/Http/Controllers/AuthController.php [hunk 1]
```diff
@@ -73,4 +74,15 @@ public function guard()
     {
         return Auth::guard();
     }
+
+    public function destroy(Request $request): JsonResponse
+    {
+        if (\Auth::user()->id === $request->get('id')) {
```
Comment (syropian) [inline app/Http/Controllers/AuthController.php]: You can just use `auth()->id()` here

File: resources/assets/js/store/modules/user.js [hunk 2]
```diff
@@ -14,6 +15,10 @@ const getters = {
 const mutations = {
   [SET_USER] (state, user) {
     state.user = user
+  },
+  [DELETE_USER] (state) {
+    state.user = {}
+    router.push('auth/logout')
```
Comment (syropian) [inline resources/assets/js/store/modules/user.js]: I generally like to keep mutations pure, so this router push call should probably be in the action after committing the mutation.

File: resources/assets/js/store/modules/user.js [hunk 3]
```diff
@@ -25,6 +30,14 @@ const actions = {
       .then(res => {
         commit(SET_USER, res)
       })
+  },
+  deleteUser ({ commit }, id) {
+    return client
+      .withAuth()
+      .delete('/api/auth/delete', { id })
+      .then(res => {
+        commit(DELETE_USER, res)
```
Comment (syropian) [inline resources/assets/js/store/modules/user.js]: After this, we should also remove their auth token out of local storage.

**Comment by syropian** [comment]
Could you also add tests please?

File: tests/Feature/AuthenticatesUsersTest.php [hunk 4]
```diff
@@ -99,4 +99,21 @@ public function a_user_can_logout()
...
+        $this->assertAuthenticated();
+
+        $id = auth()->user()->id;
+
+        $response = $this->deleteJson('/api/auth/delete', ['id' => $id])
+                        ->assertStatus(204);
+
+        $this->assertDatabaseMissing('users', ['id' => $id]);
+        $this->assertDatabaseMissing('tags', ['user_id' => $id]);
+        $this->assertDatabaseMissing('stars', ['user_id' => $id]);
```
Comment (syropian) [inline tests/Feature/AuthenticatesUsersTest.php]: To ensure stars and tags are actually getting deleted you should probably create some first :)

Comment (Williams94) [reply tests/Feature/AuthenticatesUsersTest.php]: Ahh yes, my bad!

**Comment by syropian** [comment]
That one test in `StarTest.php` seems to intermittently fail on Travis, but I'm not sure why. I'll have to look into it. Anyway, this is looking good! Feel free to merge, I'll do further UI work in a separate branch.

**Comment by syropian** [comment]
I forced a rebuild and now all the tests are passing.

**Pull request merged**